From afe15244db895dce804ae494161c82e62644ed95 Mon Sep 17 00:00:00 2001
From: Erez Shaul <eshaul@habana.ai>
Date: Sun, 08 Sep 2024 13:31:21 +0300
Subject: [PATCH] [FSW-2067] fs1: use get_64bit_addr_from_32bit_addr from com-mod

This function was moved to zephy-modules/common therefore it can
be used from there.

Change-Id: I7b33bdda8b589886d5ff01c64a237fdb051954ff
Signed-off-by: Erez Shaul <eshaul@habana.ai>
---

diff --git a/arch/fs1/arc/common.c b/arch/fs1/arc/common.c
index d3307f7..4a836db 100644
--- a/arch/fs1/arc/common.c
+++ b/arch/fs1/arc/common.c
@@ -48,6 +48,7 @@
 #include <efuse.h>
 #include <boot_desc.h>
 #include <address_space.h>
+#include <lib_common.h>
 #include "common.h"
 #include "hl.h"
 #include "auto_gen_pcie_phy_csr_locations.h"
@@ -515,27 +516,6 @@
 	desc->desc_ver = desc_ptr->desc_ver;
 	desc->crc32 = desc_ptr->crc32;
 	memcpy(desc->chksum, desc_ptr->chksum, ARRAY_SIZE(desc->chksum));
-
-	return 0;
-}
-
-/* based on lkd_dev_spec_get_img_addr from preboot */
-static u64 get_64bit_addr_from_32bit_addr(uintptr_t base)
-{
-	if (base >= IO0_SP_SRAM_BASE && base < IO0_SP_SRAM_BASE + IO_SP_SRAM_SIZE)
-		return (base & ~IO0_SP_SRAM_BASE) + IO0_SP_SRAM_FULL_BASE;
-
-	if (base >= IO0_FW_SRAM_BASE && base < IO0_FW_SRAM_BASE + IO_FW_SRAM_SIZE)
-		return (base & ~IO0_FW_SRAM_BASE) + IO0_FW_SRAM_FULL_BASE;
-
-	if (base >= IO0_MNG_ICCM_BASE && base < IO0_MNG_ICCM_BASE + IO_MNG_ICCM_SIZE)
-		return (base & ~IO0_MNG_ICCM_BASE) + IO0_MNG_ICCM_FULL_BASE;
-
-	if (base >= 0 && base < IO_MNG_ICCM_SIZE)
-		return (base & ~IO0_MNG_ICCM_BASE) + IO0_MNG_ICCM_FULL_BASE;
-
-	preboot_puts_raw(__func__);
-	preboot_puts(": error - unsupported addr\n");
 
 	return 0;
 }
